Patient Registry System/PatientRegistrySystem.DB/Repos/DbGenericRepository/DbGeneric.cs

```diff
@@ -0,0 +1,249 @@
+                    .AsNoTracking().Where(s => (s.Status != 1)).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Prescription))
+            {
+                var result = _mapper.Map<List<PrescriptionDto>>(await _applicationIdentityDbContext.Prescription
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Company))
+            {
+                var result = _mapper.Map<List<CompanyDto>>(await _applicationIdentityDbContext.Company
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Medicine))
+            {
+                var result = _mapper.Map<List<MedicineDto>>(await _applicationIdentityDbContext.Medicine
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public async Task<List<DomainEntity>> FindAllEntitiesAsync(int entityId)
+        {
+            if (typeof(DbEntity) == typeof(Record))
+            {
+                var result = _mapper.Map<List<RecordDto>>(await _applicationIdentityDbContext.Record
+                    .AsNoTracking().Where(s => (s.Status != 1) && (s.ApplicationUserId == entityId)).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Prescription))
+            {
+                var result = _mapper.Map<List<PrescriptionDto>>(await _applicationIdentityDbContext.Prescription
+                    .AsNoTracking().Where(s => s.PrescriptionId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Company))
+            {
+                var result = _mapper.Map<List<CompanyDto>>(await _applicationIdentityDbContext.Company
+                    .AsNoTracking().Where(s => s.CompanyId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Medicine))
+            {
+                var result = _mapper.Map<List<MedicineDto>>(await _applicationIdentityDbContext.Medicine
+                    .AsNoTracking().Where(s => s.MedicineId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public async Task<DomainEntity> FindEntityAsync(int entityId)
+        {
+            if (typeof(DbEntity) == typeof(Record))
```

Comment:
same comment as above. and actually, I don't you would need FindAllEntitiesAsync, because you are retrieving data by the PK, so only one record should be returned, so you can remove that method and keep FindEntityAsync.